### PR TITLE
Improvements to logging and error handling

### PR DESCRIPTION
**Pull request by FrostyX**
I am trying to figure out why this issue is happening
https://pagure.io/fedora-infrastructure/issue/11829
and come either with a fix or something that helps us debug it.

Please note that this is my first PR to this project and I have zero experience with STOMP. Proper testing before deploying is advised :D

### CODE REVIEW

File: bugzilla2fedmsg/consumer.py [hunk 1]
```diff
@@ -117,8 +118,11 @@ def setup_heartbeat(self):
         def _send_heartbeat():
             if not self._running:
                 return
-            self.stomp.beat()
-            self.setup_heartbeat()
+            try:
+                self.stomp.beat()
+            except StompConnectionError:
+                LOGGER.info("Recreating heartbeat")
+                self.setup_heartbeat()
```
Comment (FrostyX) [inline bugzilla2fedmsg/consumer.py]: I think this is not correct. My queue is not being successfully processed. I will need to take a look at this again

**Comment by FrostyX** [comment]
I think where the issue with https://pagure.io/fedora-infrastructure/issue/11829 is. Or at least may be.

Please take a look at the following code from `consumer.py`


```python
try:
    self.relay.on_stomp_message(body, frame.headers)
except Exception as ex:
    LOGGER.exception(ex)
    self.stomp.nack(frame)
else:
    self.stomp.ack(frame)
```

The `self.relay.on_stomp_message` is responsible for sending a message to fedora-messaging. Internally it calls `fedora_messaging.api.publish` which has a default timeout 30 seconds. If you try to emulate a laggy connection and do:

```python
try:
    import time
    time.sleep(30)
    self.relay.on_stomp_message(body, frame.headers)
except Exception as ex:
    LOGGER.exception(ex)
    self.stomp.nack(frame)
else:
    self.stomp.ack(frame)
```

it suddenly stops processing the queue. I've spent a couple of hours on this and can't figure out how to fix this. Can you please take a look @abompard if you spot some obvious error?

**Comment by FrostyX** [comment]
Regardless of the previous comment, can we get this reviewed and deployed to production, please? It improves logging, so we may find that the issue is somewhere else.

File: bugzilla2fedmsg/__init__.py [hunk 2]
```diff
@@ -41,7 +41,3 @@ def cli(config):
         except KeyboardInterrupt:
             consumer.stop()
             raise
-
-
-if __name__ == '__main__':
-    cli()
```
Comment (FrostyX) [inline bugzilla2fedmsg/__init__.py]: This wasn't a left-over debug line :D

Comment (LecrisUT) [reply bugzilla2fedmsg/__init__.py]: Well, good incentive to migrate to `[project.scripts]` :D

```toml
[project.scripts]
bugzilla2fedmsg = "bugzilla2fedmsg:cli"
```

(not sure if `__init__` is also necessary there)

Comment (abompard) [reply bugzilla2fedmsg/__init__.py]: Oh sorry, well, it's a bit redundant with the `console_scripts` then. I assume you added it there to test it without installing it.
Anyway I've migrated the packaging bits to poetry, to be more aligned with our other apps, and it's now a script there as well.

**Comment by abompard** [comment]
So, it's deployed in prod now. Hopefully we'll get more info next time it's stuck. Thanks!

**Comment by FrostyX** [comment]
And hopefully, it won't break even more :D
Thank you for deploying.

**Pull request merged**